dali/util/shared_mem.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_SHARED_MEM_H_
+#define DALI_UTIL_SHARED_MEM_H_
+
+#include <stdint.h>
+#include <memory>
+#include "dali/core/common.h"
+
+namespace dali {
+namespace python {
+
+/**
+ * @brief Creates or simply stores (if provided) file descriptor used for identifying
+ * shared memory chunk. Closes file descriptor in the desctructor if it
+ * wasn't closed earlier.
+ */
+class DLL_PUBLIC ShmFdWrapper {
+ public:
+  /**
+   * @brief Wrap a file descriptor, fd shouldn't be -1.
+   */
+  explicit ShmFdWrapper(int fd);
+
+  /**
+   * Create new file descriptor that can be used for shared memory chunk.
+   */
+  ShmFdWrapper();
+
+  ~ShmFdWrapper();
+
+  DISABLE_COPY_MOVE_ASSIGN(ShmFdWrapper);
```

Comment:
1. Move constructor and assignment seem quite useful.
2. This looks a lot like `unique_resource`

Reply:
Adjusted to `unique_resource`